core/cmd/local_client_test.go

```diff
@@ -265,3 +272,157 @@ func TestClient_LogToDiskOptionDisablesAsExpected(t *testing.T) {
 		})
 	}
 }
+
+func TestClient_RebroadcastTransactions_WithinRange(t *testing.T) {
+	beginningNonce := uint(7)
+	endingNonce := uint(10)
+	gasPrice := big.NewInt(100000000000)
+	gasLimit := uint64(3000000)
+	set := flag.NewFlagSet("test", 0)
+	set.Bool("debug", true, "")
+	set.Uint("beginningNonce", beginningNonce, "")
+	set.Uint("endingNonce", endingNonce, "")
+	set.Uint64("gasPriceWei", gasPrice.Uint64(), "")
+	set.Uint64("gasLimit", gasLimit, "")
+	c := cli.NewContext(nil, set, nil)
+
+	tests := []struct {
+		dbName string
+		nonce  uint
+	}{
+		{"range_start", beginningNonce},
+		{"mid_range", beginningNonce + 1},
+		{"range_end", endingNonce},
+	}
+	for _, test := range tests {
+		t.Run(test.dbName, func(t *testing.T) {
+			// Use the a non-transactional db for this test because we need to
+			// test multiple connections to the datbase, and changes made within
+			// the transaction cannot be seen from another connection.
+			config, tempDB, cleanup := cltest.BootstrapThrowawayORM(t, fmt.Sprintf("rebroadcast_txs_%s", test.dbName))
+			defer cleanup()
+			require.NoError(t, tempDB.RawDB(func(db *gorm.DB) error { return migrations.Migrate(db) }))
+			config.Config.Dialect = orm.DialectPostgres
+			connectedStore, connectedCleanup := cltest.NewStoreWithConfig(config)
+			defer connectedCleanup()
+
+			account, err := connectedStore.KeyStore.NewAccount(cltest.Password)
+			tx := cltest.CreateTxWithNonceAndGasPrice(t, connectedStore, account.Address, 0, uint64(test.nonce), 1000)
+			bumpedRaw := cltest.NewHash().Bytes()
+			bumpedRawHash := cltest.NewHash()
+			j := cltest.NewJobWithWebInitiator()
+			require.NoError(t, connectedStore.CreateJob(&j))
+			jr := cltest.CreateJobRunWithStatus(t, connectedStore, j, models.RunStatusPendingOutgoingConfirmations)
+			tx.SurrogateID = null.StringFrom(jr.ID.String())
+			require.NoError(t, connectedStore.SaveTx(tx))
+
+			// Use the same config as the connectedStore so that the advisory
+			// lock ID is the same. We set the config to be Postgres Without
+			// Lock, because the db locking stratgey is decided when we
+			// initialize the store/ORM. We set it back after initialization
+			// and then as a makeshift test, check that it was set back to
+			// WithoutLock at the end of the test.
+			config.Config.Dialect = orm.DialectPostgresWithoutLock
+			store, cleanup := cltest.NewStoreWithConfig(config)
+			config.Config.Dialect = orm.DialectTransactionWrappedPostgres
```

Comment:
Why is it necessary to switch it back to the transaction wrapped version?

Reply:
It isn't necessary, but I was checking at the end of the test that the config did get updated. I guess the behavior of the test covers that already with the fact that the records get updated.

Reply:
Ok, it turns out it is necessary, because otherwise the test would pass without the line where we actually set the dialect to `WithoutLock`. It's a makeshift test, to get around some setup/mocking constraints. I rearranged the sequence and comments in the test to try and make it more clear why we're doing it.